Assets/Scripts/Environment/Hangar/Battery/BatterySlotPlacement.cs

```diff
@@ -0,0 +1,40 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BatterySlotPlacement : InteractableObject
+{
+    public Transform Slot;
+
+    private BatteryInteractable batteryInteractableInSlot;
+
+    public void Update()
+    {
+        if(!PlayerManager.IsPaused
+          && IsPlayerColliding
+          && Input.GetKeyDown(PlayerConstants.ActionKey))
+        {
+            if (BatteryInteractable.PlayerIsHoldingABattery == true && batteryInteractableInSlot == null)
+            {
+                batteryInteractableInSlot = HangarManager.currentBatteryBeingHeld.BatteryInteractable;

```

Comment:
This reference could be null, so we should do a check here I reckon. 

Would need to check both the Wrapper and the Interactable, as either one would throw an exception. 

```
  if (HangarManager.currentBatteryBeingHeld != null 
      && HangarManager.currentBatteryBeingHeld.BatteryInteractable != null)
  {

  }
```

Reply:
Do we need to check both. Isn't just making sure HangarManager.currentBatteryBeingHeld.BatteryInteractable != null enough? Because if HangarManager.currentBatteryBeingHeld.BatteryInteractable != null, then HangarManager.currentBatteryBeingHeld can't be null, or I am missing something?

Reply:
Yeah we have to check both. 

Unless we use null propagation (the ``?`` operator). But Unity advises against doing that for MonoBehaviours. 

Basically if ``HangarManager.currentBatteryBeingHeld`` is null, and then you try to access a property on it (e.g. you do``.BatteryInteractable``, you are trying to refer to a property on something that doesn't exist. It's like saying ``null.BatteryInteractable``. ``null`` isn't an object. So the runtime can't understand that expression. 

An example here: 
![image](https://user-images.githubusercontent.com/61809127/120844950-a7a36380-c55f-11eb-864f-cc859a85ca5b.png)

https://docs.microsoft.com/en-us/dotnet/api/system.nullreferenceexception?view=net-5.0

Reply:
Ok I see, that makes sense now. The computer doesn't think as we do. It just executes a series of steps.